salt/modules/boto_route53.py

```diff
@@ -385,12 +391,15 @@ def get_record(name, zone, record_type, fetch_all=False, region=None, key=None,
             break  # the while True
 
         except DNSServerError as e:
-            # if rate limit, retry:
-            if retry_on_rate_limit and 'Throttling' == e.code:
-                log.debug('Throttled by AWS API.')
-                time.sleep(2)
+            if retry_on_rate_limit:
```

Comment:
Yeah, I agree that something like `retry_on_error` makes more sense for the way you have added the prior request handling. If you wanted to start deprecating the `retry_on_rate_limit` and `rate_limit_retries` for some more generic terms, there are docs on how to do that [here](https://docs.saltstack.com/en/latest/topics/development/deprecations.html). Since `develop` is now slated to be the `Oxygen` release, the version these kwargs should be removed is `Neon`.

From now until Neon, both syntaxes should work, with a warning to users if the have the old syntax. Then in when we're about to release Neon, the old syntax will be removed.